src/host/getset.cpp

```diff
@@ -1935,9 +1935,10 @@ void DoSrvPrivateRefreshWindow(_In_ const SCREEN_INFORMATION& screenInfo)
         sanitized.reserve(title.size());
         for (size_t i = 0; i < title.size(); i++)
         {
-            if (title.at(i) >= UNICODE_SPACE)
+            const auto ch = title.at(i);

```

Comment:
Can you convert this to a `for (const auto ch : title)` loop? The exception-checked

On top of that I'd _personally_ do it this way - it reduces binary size by half:
```
std::wstring sanitized{title};
sanitized.erase(std::remove_if(sanitized.begin(), sanitized.end(), [](auto ch) -> bool {
    return ch < 0x20 || (ch > 0x7f && ch < 0xa0);
}), sanitized.end());
return sanitized;
```
`push_back` and `emplace_back` are very expensive once they're inlined.

Reply:
I'm fine with James keeping the code the way that it was (it's not his mess to clean up!), but if he wants to fix it all the more power to him. :smile:

Reply:
Yeah, I like that approach. Not that thrilled with the code health bot's formatting, but at least it's more efficient now. Thanks @lhecker.

Reply:
I guess this range stuff gets a bit neater with C++20, but I'm assuming we're not using that yet?

Reply:
> I guess this range stuff gets a bit neater with C++20, but I'm assuming we're not using that yet?

alas no, we can't use c++20 in the OS build system quite yet ☹️